File: longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size()==0) return 0;
        sort(nums.begin(), nums.end());

        int longest = 0;
        int current = 0;
        for (int i = 0; i<nums.size()-1; i++){
            if (nums[i] == nums[i+1]) continue;
            else if (nums[i]+1==nums[i+1]) current++;
            else current = 0;
            if (current > longest) longest = current;
        }

        return longest+1;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st = set(nums)
        
        ans = 0

        for i in nums:
            if i-1 not in st:
                lmax = 1 # local max
                while i+1 in st:
                    lmax += 1
                    i += 1
                ans = max(ans, lmax)
        
        return ans